include/seqan3/core/algorithm/configuration_element_parallel_mode.hpp

```diff
@@ -0,0 +1,33 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::parallel_mode.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
+
+namespace seqan3::detail
+{
+/*!\brief A global configuration type used to enabled debugging of algorithms.
+ * \ingroup alignment_configuration
+ * \tparam wrapped_config_id_t The algorithm specific configuration id wrapped in a std::integral_constant.
+ *
+ * \details
+ *
+ * This type is used to enable the parallel mode of the algorithms.
+ */
+template <typename wrapped_config_id_t>
+struct parallel_mode : public pipeable_config_element<parallel_mode<wrapped_config_id_t>, uint32_t>
```

Comment:
What does the `uint32_t` mean at the end?

Reply:
The `pipeable_config_element` is a CRTP base that holds a single member variable called `value` that is of the type of the second template argument (in this case `uint32_t`). So it means that the configuration element will take one integer parameter (the number of threads, this is documented in the actual public configurations `align_cfg::parallel` and `search_cfg::parallel`)